app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt

```diff
@@ -8,33 +8,66 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.wire.android.appLogger
 import com.wire.android.navigation.EXTRA_CONNECTED_STATUS
+import com.wire.android.navigation.EXTRA_USER_DOMAIN
 import com.wire.android.navigation.EXTRA_USER_ID
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItem
 import com.wire.android.navigation.NavigationManager
+import com.wire.android.util.EMPTY
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.conversation.CreateConversationResult
+import com.wire.kalium.logic.feature.conversation.GetOrCreateOneToOneConversationUseCase
+import com.wire.kalium.logic.feature.publicuser.GetKnownUserUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-
 @HiltViewModel
 class OtherUserProfileScreenViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
-    private val navigationManager: NavigationManager
+    private val navigationManager: NavigationManager,
+    private val getOrCreateOneToOneConversation: GetOrCreateOneToOneConversationUseCase,
+    private val getKnownUserUseCase: GetKnownUserUseCase
 ) : ViewModel() {
 
     var state: OtherUserProfileState by mutableStateOf(OtherUserProfileState())
 
-    init {
-        savedStateHandle.get<String>(EXTRA_USER_ID)?.let { id ->
-            appLogger.d("user id $id")
-            //TODO: here we need to get the user from the Kalium
-        }
+    private val userId = UserId(
+        value = savedStateHandle.get<String>(EXTRA_USER_ID)!!,
+        domain = savedStateHandle.get<String>(EXTRA_USER_DOMAIN)!!
+    )
 
+    init {
         //TODO: internal is here untill we can get the ConnectionStatus from the user
         // for now it is just to be able to proceed forward
         val internalStatus = savedStateHandle.get<String>(EXTRA_CONNECTED_STATUS)
 
         val booleanStatus = internalStatus == "true"
 
+        if (booleanStatus) {
```

Comment:
can we rename to `isAKnownUser` or `isConnected`, `isAConnection` ?

Reply:
sure, it will be removed later one, maybe I should not add it at first, will rename it